File: Part8/C++/FunctionConcepts/FunctionConcepts_12.cpp
// Computing factorial with loop
#include <iostream>
#include <cstdio>
using namespace std;

int factorial ( int FactorialNumber_Par )
{
    int totalFactorial = 1;
    
    for ( int factorialNumber = 1; factorialNumber <= FactorialNumber_Par; factorialNumber++ )
    {
        totalFactorial = totalFactorial * factorialNumber;
    }
    
    return totalFactorial;
}

int main ()
{
    int factorialResult = factorial ( 10 );
    printf ( "Factorial of 10 is: %d\n", factorialResult );
    return 0;
}

File: Part8/Rust/FunctionConcepts/FunctionConcepts_12.rs
// Computing factorial with loop
fn factorial ( factorial_number_par: i32 ) -> i32
{
    let mut total_factorial = 1;
    
    for factorial_number in 1..=factorial_number_par
    {
        total_factorial = total_factorial * factorial_number;
    }
    
    return total_factorial;
}

fn main ()
{
    let factorial_result = factorial ( 10 );
    println! ( "Factorial of 10 is: {}", factorial_result );
}
